src/main/java/jenkins/plugins/git/GitSCMMatrixUtil.java

```diff
@@ -0,0 +1,30 @@
+package jenkins.plugins.git;
+
+import hudson.matrix.MatrixBuild;
+import hudson.matrix.MatrixRun;
+import hudson.model.Run;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.Revision;
+import hudson.plugins.git.util.Build;
+import hudson.plugins.git.util.BuildData;
+
+import java.util.Collections;
+import java.util.Set;
+
+public class GitSCMMatrixUtil {
+    public static Set<Revision> populateCandidatesFromMatrixBuild(Run build, GitSCM scm) {
+        // every MatrixRun should build the same marked commit ID
+        if (build instanceof MatrixRun) {
+            MatrixBuild parentBuild = ((MatrixRun) build).getParentBuild();
```

Comment:
Unnecessary to do this cast. You can just call `getRootBuild`, defined in core.

Reply:
That's on `AbstractBuild`, not `Run`, so wouldn't really make this noticeably easier (and I'm not sure going to the root build in all cases rather than just Matrix would be correct).